src/loginServices.js

```diff
@@ -136,12 +136,29 @@ export const getOIDCRedirectUri = (tenant, clientId) => {
   return encodeURIComponent(`${window.location.protocol}//${window.location.host}/oidc-landing?tenant=${tenant}&client_id=${clientId}`);
 };
 
-export const getTenantAndClientIdFromLoginURL = () => {
+export const getLoginTenant = (stripesOkapi, stripesConfig) => {
+  // derive from the URL
   const urlParams = new URLSearchParams(window.location.search);
+  let tenant = urlParams.get('tenant');
+  let clientId = urlParams.get('client_id');
+  if (tenant && clientId) {
+    return { tenant, clientId };
+  }
+
+  // derive from stripes.config.js::config::tenantOptions
+  if (stripesConfig.tenantOptions && Object.keys(stripesConfig?.tenantOptions).length === 1) {
+    const key = Object.keys(stripesConfig.tenantOptions)[0];
+    tenant = stripesConfig.tenantOptions[key]?.name;
+    clientId = stripesConfig.tenantOptions[key]?.clientId;
```

Comment:
Maybe a comment with the anticipated shape of the object this is accessing or describing the arbitrary(?) key i.e. `stripesConfig.tenantoptions.foo.name`